### ISSUE TITLE
having 22kohm, 3.3v or 5v?

### ISSUE THREAD

**Issue by leonardo979** [skyeperry1/Arduino_Meat_Probe_Library#4]
Hi,
I've made it on a nodeMCU board,
It works good but having 22kohm res i have to power it with 3.3v.

If I do with 5v as documented temp goes out of reasonable range.
Makes sense or I'm doing something wrong?

**Comment by NicoDupont** [comment]
Hi,

I think you have to change the arduino code If you use 5v for the meat prob because it's analog mesure not digital.

But for me 3.3v is fine and the nodemcu operate @ 3.3v.


________________________________
De : leonardo979 <notifications@github.com>
Envoyé : jeudi 7 juin 2018 11:24
À : skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library
Cc : Subscribed
Objet : [skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library] having 22kohm, 3.3v or 5v? (#4)


Hi,
I've made it on a nodeMCU board,
It works good but having 22kohm res i have to power it with 3.3v.

If I do with 5v as documented temp goes out of reasonable range.
Makes sense or I'm doing something wrong?

**Comment by bigredsmokehouse** [comment]
Hi Leonardo979,
I used this Maverick design as the basis for my propane fired smoker.  22kohms resisters work great on my 5v Arduino Mega.  
I started messing with the ESP8266 NodeMCU  board because I want to use the web page interface.  Currently I am using a Nextion display which has worked very nice.  I decided I wanted to use a smart phone interface.
I did buy some breakout boards which will give me adfitional analog ports for the nodeMCU.
Summertime keeps me outside most of the time and I do not work on the controller much.  I would love to chat with you about your adventures moving to the ESP8266 environment.   
I am planning on having a simple LCD to show the meat and ambient temps.  The recipe is created from the phone and loaded to the nodemcu which is relatively easy.  
Would love go talk with you.
BigRedSmokehouse


Sent via the Samsung Galaxy S®6 active, an AT&T 4G LTE smartphone
-------- Original message --------From: leonardo979 <notifications@github.com> Date: 6/7/18  5:24 AM  (GMT-05:00) To: skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library <Maverick-ET-73-Meat-Probe-Arduino-Library@noreply.github.com> Cc: Subscribed <subscribed@noreply.github.com> Subject: [skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library] having 22kohm,
  3.3v or 5v? (#4) 
Hi,

I've made it on a nodeMCU board,

It works good but having 22kohm res i have to power it with 3.3v.
If I do with 5v as documented temp goes out of reasonable range.

Makes sense or I'm doing something wrong?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or mute the thread.
{"@context":"http://schema.org","@type":"EmailMessage","potentialAction":{"@type":"ViewAction","target":"https://github.com/skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library/issues/4","url":"https://github.com/skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library/issues/4","name":"View Issue"},"description":"View this Issue on GitHub","publisher":{"@type":"Organization","name":"GitHub","url":"https://github.com"}}
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library","title":"skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library"}},"updates":{"snippets":[{"icon":"DESCRIPTION","message":"having 22kohm, 3.3v or 5v? (#4)"}],"action":{"name":"View Issue","url":"https://github.com/skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library/issues/4"}}}
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
"title": "having 22kohm, 3.3v or 5v? (#4)",
"sections": [
{
"text": "",
"activityTitle": "**leonardo979**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@leonardo979",
"facts": [
{
"name": "Repository: ",
"value": "skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library"
},
{
"name": "Issue #: ",
"value": 4
}
]
}
],
"potentialAction": [
{
"name": "Add a comment",
"@type": "ActionCard",
"inputs": [
{
"isMultiLine": true,
"@type": "TextInput",
"id": "IssueComment",
"isRequired": false
}
],
"actions": [
{
"name": "Comment",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueComment\",\n\"repositoryFullName\": \"skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library\",\n\"issueId\": 4,\n\"IssueComment\": \"{{IssueComment.value}}\"\n}"
}
]
},
{
"name": "Close issue",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueClose\",\n\"repositoryFullName\": \"skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library\",\n\"issueId\": 4\n}"
},
{
"targets": [
{
"os": "default",
"uri": "https://github.com/skyeperry1/Maverick-ET-73-Meat-Probe-Arduino-Library/issues/4"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 343420517\n}"
}
],
"themeColor": "26292E"
}

**Comment by leonardo979** [comment]
Hi bigredsmokehouse,
love to chat about that too!
My goal is to put on web my sample. All network part doesn't scare me, I'm gettin crazy with sampling.
My very basic approach is ESP8266 nodeMCU with single probe (ET-73) on A0 and sampling checking result on console.

First doubt is about resistor. I've found in some other projects also 10kohm res (heatermeter). I've checked that heatermeter is based on Arduino chip so 5v, but this too, and it goes with 22kohm...so my first question is: which is the right resistor for ET-73 probe? Or which is the criteria to scale it?

Apart this I see with 10kohm nodeMCU goes out of range, so I've just excluded that...

Then, applying this project and this exact schema, I'm able to sample, but I have a temperature colder than real about of 10°C.

I've tried to adjust Steinhart coeff with other versions found in the web, but i had no significant changes.

At the end I noticed that read of A0 pin, with nothing plugged, returns values around 15...17 (in scale 0,1023), meanwhile I was expecting 0

Very roughly I've subtracted 17 to analogRead(A0) before calculate 'R', and magically temperature is right.
I'm not anyway satisfied, since that approach is not "scientific"...
I don't know if my nodeMCU is damaged, or I'm doing something wrong...  

I also noticed whiled debugging that in the algorithm of average sample calculation (applied 10 sample 10 millis) first sample is often different from the rest, I've excluded that from the array in average calculation and result seems better.

Any idea?
bye
L.

**Comment by skyeperry1** [comment]
Hey leonardo979 & bigredsmokehouse,
Glad to see this library is still getting some love. leonardo979 - I've actually moved over to the NodeMCU for most of my recent projects as well and have run into the same issues when testing this library @3.3v. bigredsmokehouse - very cool idea for the recipes. I'll try and take a look at this code in the next couple weeks but summers are usually pretty busy :)

**Comment by bigredsmokehouse** [comment]
Hi Skyeperry1 & Leonardo979,

I finally get a minute to send a more detailed email.  First of all, let me preface things that an electrical engineer I am not however a programmer I am.   I have always wanted to build my own control system for a propane fired system.  Yeah, I saw a lot of things out there like the heatermeter, sp?   I have learned an amazing amount about all this microprocessor stuff.   My wife gets a kick out of my lab and the fact I am soldering boards.

I found your library and played with it on an Uno and was able to get it running.  The temps were stable.  I honestly would love to get the time to put all my stuff under this project and share it.  I need to go read on how to do this.  You two have inspired me to share everything because you are moving to the ESP8266.  That is my next step.   Here is what I built today which is running my smoker, "Big Red".

    1948 Philco Refrigerator
    Burner from https://tejassmokers.com/  (Awesome people and great to bounce ideas off.  They stand behind their products.
    Arduino Mega
    Micro SD to hold the recipes and to write logging information during the process, very important
    2 Maverick probes (they work but I do not like the 2.5mm interface, Look at the ones from Auber Instruments.  6.5mm).  I have smoked tons with no problems... I can live with the tiny connectors.
    I built a little driver to run 2 12 volt fans (2 amp max) using an LN298N
    Transformer 110v to 24 AC (to turn trigger the gas valve to open when I want)
    Robertshaw gas valve (anyone will do.  A new one ran me $70), thermocoupler was included
    Shielded cable which helped on any motor interference (i.e. gas valve)
    RJ45 - Ethernet interfaces.  Everyone has cables laying around.
    Separate 5v source
    12V wall wart to power the fans.  One fan is for the cold smoke unit and the other one is to pull the smoke out of my vent.  Both of these are settings that you can control the speed of the fans.  I am using PWM on the LN298N
    Nextion 3.5" display.  Although it was a pain but once I figured things out coding wise, It is a pretty cool touch screen.  It talks to the Arduino and visa versa.

I usually just enter my formula.  It can be 4 steps and based on time or meat temperature.  For example, Run for 1 hour at 180 degrees, then 1.5 hours at 190, bump it up to 200 until the meat temps hit 160.  It works beautifully.  All processing is logged.

 I have been messing with the NodeMCU boards and I am convinced that is the way to go.  I want the smart phone interface and eliminate the Nextion display.  I was going to integrate the Mega with the ESP but just too much of a pain to do it.  I am choosing to replace my Arduino.  I bought a couple of these ADS1115 breakouts and found some really good videos on how to easily use this to expand the analog ports on the nodemcu.

I ordered a Max31865 breakout board to consider RTD type of probes.  I keep reading about how great the PT1000 probes are.  My gut tells me I have not had a single problem with the maverick probes.  I was going to see if I can interface this to the ADS1115 board.

My SD breakout should work fine on the ESP8266.

I can build the HTML interface pages pretty easy.  I am going to set it up as an access point (AP) so I don't have to mess with the internet.  If I had the right people helping on this one, I would do this as total IoT.    All logging info (temps) is being tracked

In summary, I would love to have a few people to put our heads together and build a stable nodemcu process as I described.   I would love to hear back from you.

![image](https://user-images.githubusercontent.com/24206045/41786185-cf4b239a-7612-11e8-873b-df71600e6a51.png)

![image](https://user-images.githubusercontent.com/24206045/41786212-e596624a-7612-11e8-81d9-03f5699fbd44.png)

**Comment by mixographer** [comment]
I think the nodeMCU uses 12-bits for the analog input. This means that the analog values read at 4095 at full 3.3v. So I think you wouldn't change the resistor, but rather change the code. I think the lines in question are in the getTemp_ functions. Change the 1024 to 4096 and see if this is the case? I can't try it today, but I will try with a esp32 dev board.

**Comment by leonardo979** [comment]
I'm using ESP8266, and input is 10bit.
Apart this, tested again trying some configurations, (and another ESP board!), I'm still having background noise in analog input with device completely disconnected...I still see a random value between 5 and 12 as reading.
I also seem that wifi actions have impacts on pin reading. I've applied a http call to store sample in a server, and noticed that with wifi disabled and all that part commented, all analog samples are stable (920,920,920.....). With wifi first samples are little higher(924, 922, 920, 920, 920...)

In the end...right now it's having a "reasonable" behavior, but the temp is 5°c over the real one (taken with 2 different real commercial thermometer)... problem is...i don't know now how to adjust this in the right way

**Comment by phreaknes** [comment]
Did we get a solve to this? I have it working perfectly on an 5v Uno but need on a ESP32. I think the voltage dividing resistor is the culprit here. I also from the [data sheet for the ET-73](https://drive.google.com/file/d/1ukcaFtORlLmLLrnIlCA0BvS1rEwbFoyd4ReqIFV8y3iL1sojljPAW8x8bYZW/view) probes if that helps.

**Comment by ve7kilohertz** [comment]
Hi All, Has anyone updated the code or a way to use 3.3 instead of 5V?  I have it working with 3.3V but the readings above 100F start to increase to 233F @ a real temp of 212F.  Using NODEMCU, the ET73 and 22K R. Thx!
